### ISSUE TITLE
Incorrect parse error on pattern matching using 'var'

### ISSUE THREAD

**Issue by cajun-rat** [javaparser/javaparser#4613]
The following compiles under Java 21 but is rejected by javaparser with `(line 5,col 28) "var" is not allowed here.`

```
public class Main {
  public record Point(int x, int y) {}
  public static void main(String[] args) {
    var p = new Point(1,2);
    if (p instanceof Point(var x, var y)) {
      System.out.println("yes");
    }
  }
}
```

I believe it is valid Java 21 code: it compiles successfully with `javac --source 21 Main.java` and the Java Language Specification appears to allow it:

- LocalVariableType can contain `var` [ref](https://docs.oracle.com/javase/specs/jls/se21/html/jls-14.html#jls-LocalVariableType)
- Pattern is recursively either a RecordPattern or a LocalVariableDeclaration, which is a LocalVariableType plus a variable name [ref](https://docs.oracle.com/javase/specs/jls/se21/html/jls-14.html#jls-14.30.1)

I saw a related ticket https://github.com/javaparser/javaparser/issues/4550, but this is a different issue. In my example changing `var` to `int` on 5 (but not line 4) causes it to parse successfully.

**Comment by gpoussel** [comment]
Hello,

It looks like I cannot reproduce your issue. The following test case passes on my side:
```java
ParserConfiguration cfg = new ParserConfiguration();
cfg.setLanguageLevel(ParserConfiguration.LanguageLevel.JAVA_21);
JavaParser javaParser = new JavaParser(cfg);
ParseResult<CompilationUnit> parseResult = javaParser.parse("public class Main {"
        + EOL + "   public record Point(int x, int y) {}"
        + EOL + "   public static void main(String[] args) {"
        + EOL + "        var p = new Point(1,2);"
        + EOL + "        if (p instanceof Point(var x, var y)) {"
        + EOL + "             System.out.println(\"yes\");"
        + EOL + "        }"
        + EOL + "    }"
        + EOL + "}");
assertEquals(0, parseResult.getProblems().size());
```

Same result on 3.26.0, 3.26.1 and 3.26.2

**Comment by cajun-rat** [comment]
Thank you for looking to this. I get a failure with the following example:


`# cat Demo.java`
```
import com.github.javaparser.JavaParser;
import com.github.javaparser.ParseProblemException;
import com.github.javaparser.ParseResult;
import com.github.javaparser.ParserConfiguration;
import com.github.javaparser.ast.CompilationUnit;

public class Demo {
  public static String EOL = "\n";
  public static void main(String[] args) {
    ParserConfiguration cfg = new ParserConfiguration();
    cfg.setLanguageLevel(ParserConfiguration.LanguageLevel.JAVA_21);
    JavaParser javaParser = new JavaParser(cfg);
    ParseResult<CompilationUnit> parseResult = javaParser.parse("public class Main {"
        + EOL + "   public record Point(int x, int y) {}"
        + EOL + "   public static void main(String[] args) {"
        + EOL + "        var p = new Point(1,2);"
        + EOL + "        if (p instanceof Point(var x, var y)) {"
        + EOL + "             System.out.println(\"yes\");"
        + EOL + "        }"
        + EOL + "    }"
        + EOL + "}");
    System.out.println(parseResult);
  }
}
```

```
java -cp javaparser-core-3.26.2.jar Demo.java
Parsing failed:
(line 5,col 32) "var" is not allowed here.
(line 5,col 39) "var" is not allowed here.
```

I downloaded the jar manually from https://mvnrepository.com/artifact/com.github.javaparser/javaparser-core/3.26.2 It matches the version that `mvn` downloaded and cached in the `.m2`  directory.

```
# sha1sum javaparser-core-3.26.2.jar
795eca30f20fdc110d1a02a1f0e7ad207e17e7c1  javaparser-core-3.26.2.jar
```
(This matches `javaparser-core-3.26.2.jar.sha1`)


I'm guessing we have different versions of a thing called 'javaparser v3.26.2'. Can you confirm the true checksum of the release, please?

**Comment by gpoussel** [comment]
Whoops, sorry... I was testing the current development `master` branch and never properly switched to mentionned versions; apologies for the confusion.
However, after a double check, the issue has been fixed on `master` with 602f8edcf651b3ea6a08e1e400fc9922018b16d1 (issue #4579, PR #4581).

**Comment by cajun-rat** [comment]
Thank you! I can confirm it is fixed on the latest master:

```
java -cp javaparser-core-3.26.3-SNAPSHOT.jar Demo.java
Parsing successful
```

Cheers!
